LibOS/shim/src/shim_malloc.c

```diff
@@ -338,6 +338,8 @@ void * malloc (size_t size)
     void * mem = slab_alloc(slab_mgr, size);
 #endif
 
+    assert(mem != (void *) 0xdeadbeef);
```

Comment:
What is the purpose of this? 0xdeadbeef is a perfectly valid memory address (unaligned, but still...). I know it's also a SLAB canary, but why do you compare the returned address with it?

Reply:
Also, the code will break after changing this magic value in the slab allocator, you should use the value defined there.

Reply:
Shoot... this is my fault. I will remove it.